apps/web/core/hooks/common/use-has-mounted.ts

```
@@ -11,3 +11,35 @@ export const useHasMounted = () => {
 };
+
+/**
+ * Custom hook that executes a callback on dependency changes with conditional first render execution.
+ * The condition is evaluated dynamically on each render.
+ *
+ * @param callback - The function to execute when dependencies change
+ * @param dependencies - Array of dependencies to watch for changes
+ * @param shouldSkipFirstRender - Whether to skip the first render
+ */
+export const useConditionalUpdateEffect = (
+	callback: () => void | (() => void),
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)